### DISCUSSION TITLE
Template script for camera3D mouse look-around in 4.0?

### DISCUSSION THREAD

**Discussion by elvisish** [godotengine/godot-proposals#5601] [Scripting]
The CharacterBody3D template script is a really neat way to get up and running (literally) with moving around, it would be great if the Camera3D node had a mouse look-around template as well for any first-person game work.

**Comment by Calinou** [comment]
Should the template script be part of Camera3D (for freelook cameras not related to a character), or part of CharacterBody3D? If you want an actual FPS character, you need the latter.

**Reply by elvisish** [reply]
> Should the template script be part of Camera3D (for freelook cameras not related to a character), or part of CharacterBody3D? If you want an actual FPS character, you need the latter.

That's true, for the directional rotation, I hadn't considered that. Perhaps if a camera is present as part of the node hierarchy (search for a camera node child) and the CharacterBody3D script finds it, the look function would be enabled?
